### PR TITLE
opae-sdk: fix debug build gcc 12 errors

### PR DESCRIPTION
**Pull request by anandaravuri**
- fix debug build gcc 12 build errors  on fedora 37
- fix  debug build  clang errors
- fix  debug build  scan-build errors

Signed-off-by: anandaravuri <ananda.ravuri@intel.com>

### CODE REVIEW

File: binaries/fpgad/command_line.c [hunk 1]
```diff
@@ -311,7 +311,7 @@ int cmd_canonicalize_paths(struct fpgad_config *c)
 		if (snprintf(c->logfile, sizeof(c->logfile),
 			     "%s/%s", c->directory, DEFAULT_LOG) < 0) {
 			len = strnlen("./" DEFAULT_LOG,
-					sizeof(c->logfile) - 1);
+					sizeof("./"));
```
Comment (tswhison) [inline binaries/fpgad/command_line.c:314]: This should be
```c
sizeof("./" DEFAULT_LOG)
```

File: binaries/fpgad/command_line.c [hunk 2]
```diff
@@ -323,7 +323,7 @@ int cmd_canonicalize_paths(struct fpgad_config *c)
 		if (snprintf(c->logfile, sizeof(c->logfile),
 			     "%s/%s", c->directory, buf) < 0) {
 			len = strnlen("./" DEFAULT_LOG,
-					sizeof(c->logfile) - 1);
+					sizeof("./"));
```
Comment (tswhison) [inline binaries/fpgad/command_line.c:326]: This should be
```c
sizeof("./" DEFAULT_LOG)
```

File: binaries/fpgad/command_line.c [hunk 3]
```diff
@@ -350,7 +350,7 @@ int cmd_canonicalize_paths(struct fpgad_config *c)
 		if (snprintf(c->pidfile, sizeof(c->pidfile),
 			     "%s/%s", c->directory, DEFAULT_PID) < 0) {
 			len = strnlen("./" DEFAULT_PID,
-					sizeof(c->pidfile) - 1);
+					sizeof("./"));
```
Comment (tswhison) [inline binaries/fpgad/command_line.c:353]: This should be
```c
sizeof("./" DEFAULT_PID)
```

File: binaries/fpgad/command_line.c [hunk 4]
```diff
@@ -363,7 +363,7 @@ int cmd_canonicalize_paths(struct fpgad_config *c)
 		if (snprintf(c->pidfile, sizeof(c->pidfile),
 			     "%s/%s", c->directory, buf) < 0) {
 			len = strnlen("./" DEFAULT_PID,
-					sizeof(c->pidfile) - 1);
+					sizeof("./"));
```
Comment (tswhison) [inline binaries/fpgad/command_line.c:366]: This should be
```c
sizeof("./ DEFAULT_PID)
```

**Review by tswhison — COMMENTED** [review]
Please update the fpgad changes according to the comments.  Also, could you please test the fpgad binary after making the changes by running and and verifying that the paths are set correctly?

File: binaries/fpgad/command_line.c (hunk 1 above)
Comment (anandaravuri) [reply binaries/fpgad/command_line.c]: fixed

File: binaries/fpgad/command_line.c (hunk 2 above)
Comment (anandaravuri) [reply binaries/fpgad/command_line.c]: fixed

File: binaries/fpgad/command_line.c (hunk 3 above)
Comment (anandaravuri) [reply binaries/fpgad/command_line.c]: fixed

File: binaries/fpgad/command_line.c (hunk 4 above)
Comment (anandaravuri) [reply binaries/fpgad/command_line.c]: fixed

**Review by tswhison — APPROVED** [review]

**Pull request merged**